src/quickdocumentdialog.cpp

```diff
@@ -303,15 +282,65 @@ void QuickDocumentDialog::Init()
         item->setCheckState(Qt::Unchecked);
     }
 
+	QTableWidget *table = ui.tableWidgetPackages;
+	table->clear();
+	table->setColumnCount(2);
+	table->setHorizontalHeaderLabels( QStringList( { tr("Package"), tr("Description") } ) );
+	table->horizontalHeader()->setStretchLastSection(true);
+	table->setSelectionMode(QAbstractItemView::ExtendedSelection);
+	table->setSelectionBehavior(QAbstractItemView::SelectRows);
+	table->verticalHeader()->hide();
+
+	//each QStringList holds 2 items: the name of the package, and a short package description. These constitute a row of the table of the packages tab
+	QList<QStringList> packages = QList<QStringList>()
+		<< QStringList( {"amssymb"     , "Mathematical symbols from AMS"} )
+		<< QStringList( {"graphicx"    , "Graphics package, easily include images (s. Insert Graphic Wizard)"} )
+		<< QStringList( {"hyperref"    , "Support for hyperlinks in your document"} )
+		<< QStringList( {"mathtools"   , "Extension package to amsmath incl. fixes for bugs in amsmath, loads amsmath"} )
+		<< QStringList( {"amsthm"      , "Define your theorem like env., has to be loaded after amsmath"} )
+		<< QStringList( {"nameref"     , "Reference to names of chapters, sections, ..., loaded by hyperref"} )
+		<< QStringList( {"thmtools"    , "Extention package to amsthm"} )
+		<< QStringList( {"xcolor"      , "Sophisticated package for colors, with table option use colors in tables"} )
+		;
+	//add user given packages
+	for (const QString& package:otherPackagesList){
+		packages << QStringList( { package, "" } );
+	}
+	//setup packages table 
+	int row=-1;
+	int tableHeight = table->horizontalHeader()->height() + 2;
+	for (const QStringList& data:packages){
+		++row;
+		QString pkgName = data[0],
+				pkgDescription = data[1];
+
+		QTableWidgetItem *itemPkgName = new QTableWidgetItem(pkgName);
+		QTableWidgetItem *itemPkgDescription = new QTableWidgetItem(pkgDescription);
+
+		itemPkgName->setFlags(Qt::ItemIsEnabled | Qt::ItemIsSelectable | Qt::ItemIsUserCheckable);
+		itemPkgDescription->setFlags(Qt::ItemIsEnabled | Qt::ItemIsSelectable);
+
+		table->setRowCount(row+1);
+		table->setItem(row,0,itemPkgName);
+		table->setItem(row,1,itemPkgDescription);
+		tableHeight += table->rowHeight(row);
+
+		bool found = packagesUsed.contains(pkgName);
+		itemPkgName->setCheckState( found ? Qt::Checked : Qt::Unchecked );
+		updateSelected(row);
+	}
+	int height = ui.GridPackages->geometry().height();
```

Comment:
this looks odd.
Why is it necessary ?

Reply:
You mean referencing updateSelected? This ensures that the line having checkState set appears selected. Otherwise one could click in a row (say in second column) which turns the row selected (blue) but the checkbox isn't selected or in a checkbox but row is not selected. Thus this is done to avoid visual ambiguity.

Reply:
also the height stuff.
There should be automatisms for that.

Reply:
Maybe, I used hours to figure out which way could work at to try them. But nothing helped. I think the main reason for this is your which not to see a scrollbar at the beginning. If I don't specify the height, then Qt will distribute space between table and what is below the table. Now one could think change ratio but if the table widget height then is larger as needed an empty space appears after the rows but still inside the frame of the table widget. I have seen this when setting the spacer size policy to minimum. And as a side effect when clicking in this extra space all rows get unselected.

Reply:
Have you tried a vertical spacer below the table ?

Reply:
I you don't have a spacer then the table widget fits exactly the header plus rows, but it is vertically centered in the tab which looks random. So we need a spacer (I already mentioned above) below the table widget that moves the table upwards. To do so the spacer needs a policy of minimumExpanding. But this means no other than that the spacer and the table take their space in a specific ratio, normally 1:1 (to prevent this I set the minimum and maximum height of the table to the same value). But this will introduce a scrollbar because the table needs more space. Of course we could now try to approximate the real number needed as ratio by rational numbers (you need to use integers for specifying proportions). But after adding one more row (or just changing to windows style which has different decorations not counted into the heights) this will break again. I don't see how we could break out of the circle.